cloud/scope/machine.go

```diff
@@ -78,31 +79,45 @@ func NewMachineScope(ctx context.Context, apiKey string, params MachineScopePara
 	}
 
 	if credentialRef != nil {
-		data, err := getCredentialDataFromRef(ctx, params.Client, *credentialRef, defaultNamespace)
+		// TODO: This key is hard-coded (for now) to match the externally-managed `manager-credentials` Secret.
+		apiToken, err := getCredentialDataFromRef(ctx, params.Client, *credentialRef, defaultNamespace, "apiToken")
 		if err != nil {
 			return nil, fmt.Errorf("credentials from secret ref: %w", err)
 		}
-		apiKey = string(data)
+		apiKey = string(apiToken)
+
+		dnsToken, err := getCredentialDataFromRef(ctx, params.Client, *credentialRef, defaultNamespace, "dnsToken")
+		if err != nil || dnsToken == nil {
+			dnsToken = apiToken
+		}
+		dnsKey = string(dnsToken)
```

Comment:
could we default to apiKey if dnsKey is an empty string?

Reply:
done